### DISCUSSION TITLE
Lifecycle hooks on watchtower itself

### DISCUSSION THREAD

**Discussion by TrueXakeP** [containrrr/watchtower#1299] [Q&A] [answered]
Is it possible to run hooks on watchtower itself so I can do
```shell
docker run -d \
  -v /var/run/docker.sock:/var/run/docker.sock \
  --label=com.centurylinklabs.watchtower.lifecycle.pre-check="/watchtower --scope pull-only --no-restart --run-once" \
  containrrr/watchtower \
  --no-pull
```
?
Or using 'command' syntax: `... --label=com.centurylinklabs.watchtower.lifecycle.pre-check="--scope pull-only --no-restart --run-once" ...`

The question is about this line, which as how I can understand prepends the hook command with `sh -c` and makes it invalid for the watchtower image (and for other images that do not contain `sh` binary)
https://github.com/containrrr/watchtower/blob/b3b45ab19e57dc822d3341d9a586919a49d2900e/pkg/container/client.go#L370

I'm not sure that this is a bug or feature request. And where to post it.

**Comment by piksel** [comment, marked as answer]
> Is it possible to run hooks on watchtower itself so I can do

I think that would cause some problems unrelated to the `Cmd`. It's at least not considered in the implementation and wholly untested.

Regarding the `sh -c` prefix to the command, I think it *might* actually still work because of how the fallback mechanism works inside docker. There is no way for the API user to know whether the `Cmd` will actually resolve to something in the `PATH`, so it has to be implemented in the daemon somehow.
